### ISSUE TITLE
bin/deploy failure

### ISSUE THREAD

**Issue by racke** [SysPete/Icecat-Schema#3]
```
DBIx::Class::Storage::DBI::catch {...} (): DBIx::Class::Storage::DBI::try {...} (): DBI Exception: DBD::mysql::db do failed: Invalid default value for 'last_published' [for Statement "CREATE TABLE `measure` (
  `measure_id` integer NOT NULL,
  `sid` integer NOT NULL,
  `tid` integer NOT NULL,
  `sign` varchar(255) NULL,
  `updated` timestamp NOT NULL,
  `last_published` timestamp NOT NULL,
  `system_of_measurement` enum('metric', 'imperial') NOT NULL DEFAULT 'metric',
  INDEX `measure_idx_sid` (`sid`),
  INDEX `measure_idx_tid` (`tid`),
  PRIMARY KEY (`measure_id`),
  CONSTRAINT `measure_fk_sid` FOREIGN KEY (`sid`) REFERENCES `sid_index` (`sid`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `measure_fk_tid` FOREIGN KEY (`tid`) REFERENCES `tid_index` (`tid`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB"] at ./bin/deploy line 9
 (running "CREATE TABLE `measure` (
  `measure_id` integer NOT NULL,
  `sid` integer NOT NULL,
  `tid` integer NOT NULL,
  `sign` varchar(255) NULL,
  `updated` timestamp NOT NULL,
  `last_published` timestamp NOT NULL,
  `system_of_measurement` enum('metric', 'imperial') NOT NULL DEFAULT 'metric',
  INDEX `measure_idx_sid` (`sid`),
  INDEX `measure_idx_tid` (`tid`),
  PRIMARY KEY (`measure_id`),
  CONSTRAINT `measure_fk_sid` FOREIGN KEY (`sid`) REFERENCES `sid_index` (`sid`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `measure_fk_tid` FOREIGN KEY (`tid`) REFERENCES `tid_index` (`tid`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB") at ./bin/deploy line 9
DBIx::Class::Storage::DBI::catch {...} (): DBIx::Class::Storage::DBI::try {...} (): DBI Exception: DBD::mysql::db do failed: Invalid default value for 'last_published' [for Statement "CREATE TABLE `measure_sign` (
  `measure_sign_id` integer NOT NULL,
  `measure_id` integer NOT NULL,
  `langid` integer NOT NULL,
  `value` varchar(255) NOT NULL,
  `updated` timestamp NOT NULL,
  `last_published` timestamp NOT NULL,
  INDEX `measure_sign_idx_langid` (`langid`),
  INDEX `measure_sign_idx_measure_id` (`measure_id`),
  PRIMARY KEY (`measure_sign_id`),
  UNIQUE `measure_sign_measure_id_langid` (`measure_id`, `langid`),
  CONSTRAINT `measure_sign_fk_langid` FOREIGN KEY (`langid`) REFERENCES `language` (`langid`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `measure_sign_fk_measure_id` FOREIGN KEY (`measure_id`) REFERENCES `measure` (`measure_id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB"] at ./bin/deploy line 9
 (running "CREATE TABLE `measure_sign` (
  `measure_sign_id` integer NOT NULL,
  `measure_id` integer NOT NULL,
  `langid` integer NOT NULL,
  `value` varchar(255) NOT NULL,
  `updated` timestamp NOT NULL,
  `last_published` timestamp NOT NULL,
  INDEX `measure_sign_idx_langid` (`langid`),
  INDEX `measure_sign_idx_measure_id` (`measure_id`),
  PRIMARY KEY (`measure_sign_id`),
  UNIQUE `measure_sign_measure_id_langid` (`measure_id`, `langid`),
  CONSTRAINT `measure_sign_fk_langid` FOREIGN KEY (`langid`) REFERENCES `language` (`langid`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `measure_sign_fk_measure_id` FOREIGN KEY (`measure_id`) REFERENCES `measure` (`measure_id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB") at ./bin/deploy line 9
DBIx::Class::Storage::DBI::catch {...} (): DBIx::Class::Storage::DBI::try {...} (): DBI Exception: DBD::mysql::db do failed: Invalid default value for 'updated' [for Statement "CREATE TABLE `product` (
  `product_id` integer NOT NULL,
  `supplier_id` integer NOT NULL,
  `prod_id` varchar(60) NOT NULL DEFAULT '',
  `catid` integer NOT NULL DEFAULT 0,
  `user_id` integer NOT NULL DEFAULT 1,
  `launch_date` timestamp NULL,
  `obsolence_date` integer NULL,
  `name` varchar(255) NOT NULL DEFAULT '',
  `low_pic` varchar(255) NOT NULL DEFAULT '',
  `high_pic` varchar(255) NOT NULL DEFAULT '',
  `publish` char(1) NOT NULL DEFAULT 'Y',
  `public` char(1) NOT NULL DEFAULT 'Y',
  `thumb_pic` varchar(255) NULL,
  `updated` timestamp NOT NULL,
  `date_added` date NOT NULL,
  `family_id` integer NOT NULL DEFAULT 0,
  `dname` varchar(255) NOT NULL DEFAULT '',
  `topseller` varchar(255) NOT NULL DEFAULT '',
  `low_pic_size` integer NULL DEFAULT 0,
  `high_pic_size` integer NULL DEFAULT 0,
  `thumb_pic_size` integer NULL DEFAULT 0,
  `high_pic_width` integer NOT NULL DEFAULT 0,
  `high_pic_height` integer NOT NULL DEFAULT 0,
  `low_pic_width` integer NOT NULL DEFAULT 0,
  `low_pic_height` integer NOT NULL DEFAULT 0,
  `high_pic_origin` varchar(255) NOT NULL DEFAULT '',
  `series_id` integer NOT NULL DEFAULT 1,
  `checked_by_supereditor` tinyint NOT NULL DEFAULT 0,
  `medium_pic` varchar(255) NOT NULL DEFAULT '',
  `medium_pic_size` integer NOT NULL DEFAULT 0,
  `medium_pic_width` integer NOT NULL DEFAULT 0,
  `medium_pic_height` integer NOT NULL DEFAULT 0,
  `high_pic_origin_size` integer NOT NULL DEFAULT 0,
  `quality` varchar(16) NOT NULL,
  `on_market` boolean NOT NULL DEFAULT '0',
  `got_images` boolean NOT NULL,
  INDEX `product_idx_catid` (`catid`),
  INDEX `product_idx_supplier_id` (`supplier_id`),
  PRIMARY KEY (`product_id`),
  UNIQUE `product_prod_id_supplier_id` (`prod_id`, `supplier_id`),
  CONSTRAINT `product_fk_catid` FOREIGN KEY (`catid`) REFERENCES `category` (`catid`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `product_fk_supplier_id` FOREIGN KEY (`supplier_id`) REFERENCES `supplier` (`supplier_id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB"] at ./bin/deploy line 9
 (running "CREATE TABLE `product` (
  `product_id` integer NOT NULL,
  `supplier_id` integer NOT NULL,
  `prod_id` varchar(60) NOT NULL DEFAULT '',
  `catid` integer NOT NULL DEFAULT 0,
  `user_id` integer NOT NULL DEFAULT 1,
  `launch_date` timestamp NULL,
  `obsolence_date` integer NULL,
  `name` varchar(255) NOT NULL DEFAULT '',
  `low_pic` varchar(255) NOT NULL DEFAULT '',
  `high_pic` varchar(255) NOT NULL DEFAULT '',
  `publish` char(1) NOT NULL DEFAULT 'Y',
  `public` char(1) NOT NULL DEFAULT 'Y',
  `thumb_pic` varchar(255) NULL,
  `updated` timestamp NOT NULL,
  `date_added` date NOT NULL,
  `family_id` integer NOT NULL DEFAULT 0,
  `dname` varchar(255) NOT NULL DEFAULT '',
  `topseller` varchar(255) NOT NULL DEFAULT '',
  `low_pic_size` integer NULL DEFAULT 0,
  `high_pic_size` integer NULL DEFAULT 0,
  `thumb_pic_size` integer NULL DEFAULT 0,
  `high_pic_width` integer NOT NULL DEFAULT 0,
  `high_pic_height` integer NOT NULL DEFAULT 0,
  `low_pic_width` integer NOT NULL DEFAULT 0,
  `low_pic_height` integer NOT NULL DEFAULT 0,
  `high_pic_origin` varchar(255) NOT NULL DEFAULT '',
  `series_id` integer NOT NULL DEFAULT 1,
  `checked_by_supereditor` tinyint NOT NULL DEFAULT 0,
  `medium_pic` varchar(255) NOT NULL DEFAULT '',
  `medium_pic_size` integer NOT NULL DEFAULT 0,
  `medium_pic_width` integer NOT NULL DEFAULT 0,
  `medium_pic_height` integer NOT NULL DEFAULT 0,
  `high_pic_origin_size` integer NOT NULL DEFAULT 0,
  `quality` varchar(16) NOT NULL,
  `on_market` boolean NOT NULL DEFAULT '0',
  `got_images` boolean NOT NULL,
  INDEX `product_idx_catid` (`catid`),
  INDEX `product_idx_supplier_id` (`supplier_id`),
  PRIMARY KEY (`product_id`),
  UNIQUE `product_prod_id_supplier_id` (`prod_id`, `supplier_id`),
  CONSTRAINT `product_fk_catid` FOREIGN KEY (`catid`) REFERENCES `category` (`catid`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `product_fk_supplier_id` FOREIGN KEY (`supplier_id`) REFERENCES `supplier` (`supplier_id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB") at ./bin/deploy line 9
DBIx::Class::Storage::DBI::catch {...} (): DBIx::Class::Storage::DBI::try {...} (): DBI Exception: DBD::mysql::db do failed: Can't create table `icecat`.`product_feature` (errno: 150 "Foreign key constraint is incorrectly formed") [for Statement "CREATE TABLE `product_feature` (
  `product_feature_id` integer NOT NULL auto_increment,
  `product_id` integer NOT NULL,
  `category_feature_id` integer unsigned NOT NULL,
  `value` varchar(20000) NOT NULL DEFAULT '',
  `updated` timestamp NOT NULL,
  INDEX `product_feature_idx_category_feature_id` (`category_feature_id`),
  INDEX `product_feature_idx_product_id` (`product_id`),
  PRIMARY KEY (`product_feature_id`),
  UNIQUE `product_feature_category_feature_id_product_id` (`category_feature_id`, `product_id`),
  CONSTRAINT `product_feature_fk_category_feature_id` FOREIGN KEY (`category_feature_id`) REFERENCES `category_feature` (`category_feature_id`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `product_feature_fk_product_id` FOREIGN KEY (`product_id`) REFERENCES `product` (`product_id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB"] at ./bin/deploy line 9
 (running "CREATE TABLE `product_feature` (
  `product_feature_id` integer NOT NULL auto_increment,
  `product_id` integer NOT NULL,
  `category_feature_id` integer unsigned NOT NULL,
  `value` varchar(20000) NOT NULL DEFAULT '',
  `updated` timestamp NOT NULL,
  INDEX `product_feature_idx_category_feature_id` (`category_feature_id`),
  INDEX `product_feature_idx_product_id` (`product_id`),
  PRIMARY KEY (`product_feature_id`),
  UNIQUE `product_feature_category_feature_id_product_id` (`category_feature_id`, `product_id`),
  CONSTRAINT `product_feature_fk_category_feature_id` FOREIGN KEY (`category_feature_id`) REFERENCES `category_feature` (`category_feature_id`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT `product_feature_fk_product_id` FOREIGN KEY (`product_id`) REFERENCES `product` (`product_id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB") at ./bin/deploy line 9
```

**Comment by racke** [comment]
Server version: 10.0.24-MariaDB-7 Debian unstable

**Comment by racke** [comment]
I suppose you can't have two timestamp fields in that way, e.g. http://stackoverflow.com/questions/9192027/invalid-default-value-for-create-date-timestamp-field

**Comment by SysPete** [comment]
Hmmm. Not tested before with MySQL.

**Comment by racke** [comment]
Ah, I found traces of MySQL in four scripts.

**Comment by SysPete** [comment]
Fixed by 0a0ed44c7eb13886fb2ebe99440875c255205e1f (timestamp issue) and 68627a32c3c66bdc208c71621b203bd7d1406296 (FK issue). Thanks @racke!
